src/EventListener/BootstrapLegacyListener.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2015 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Bootstraps the legacy Contao environment
+ *
+ * @author Christian Schiffler <https://github.com/discordier>
+ * @author Yanick Witschi <https://github.com/toflar>
+ */
+class BootstrapLegacyListener
+{
+    /**
+     * The router service to retrieve routes from.
+     *
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * The kernel root directory.
+     *
+     * @var string
+     */
+    private $rootDir;
+
+    /**
+     * Create a new instance.
+     *
+     * @param RouterInterface $router  The router service to retrieve routes from.
+     * @param string          $rootDir The kernel root directory.
+     */
+    public function __construct(RouterInterface $router, $rootDir)
+    {
+        $this->router  = $router;
+        $this->rootDir = $rootDir;
+    }
+
+    /**
+     * Boot the legacy code for all web SAPIs.
+     *
+     * @param GetResponseEvent $event The event object
+     *
+     * @throws RouteNotFoundException When no route could be found for the request.
+     */
+    public function onBootLegacyForRequest(GetResponseEvent $event)
```

Comment:
So was I thinking.
We might want to add a phpDoc stating that it will get called on the event `kernel.request` but this might be superfluous information.

If we end up naming the method after the event, we must also change: `onBootLegacyForConsole` to `onConsoleCommand` which I personally think of misleading.